### ISSUE TITLE
3e class abilities

### ISSUE THREAD

**Issue by gizmomathboy** [gizmomathboy/hyperborea_apps#9]
New stuff like Enlist Henchmen and New Weapon.

Make sure class abilities are updated.

**Comment by Sarsem** [comment]
Assassin issues all resolved

**Comment by Sarsem** [comment]
Barbarian issues all resolved

**Comment by Sarsem** [comment]
Bard issues all resolved

**Comment by gizmomathboy** [comment]
I think I have them sorted out except for the Illusionist spells for the bard. The ability stuff was just some logic problems and data copying gone wrong...along with some pdf-centric code that I need to push elsewhere. That's a longer term refactoring

**Comment by gizmomathboy** [comment]
Hey, I think I sorted out the bard spells/day stuff. Along with the other split school classes and such

**Comment by Sarsem** [comment]
Assassin and Barbarian look good now; Bard is still missing Hide and Scroll Writing, and Countersong misspells "opponents"

**Comment by Sarsem** [comment]
Berserker issues all resolved

**Comment by gizmomathboy** [comment]
Ok...found the problem with abilities and they should be listed now.

Berserkers have Extraordinary ST, CN

Maybe you are thinking of Barbarians?

**Comment by Sarsem** [comment]
Yes, my mistake on the Extraordinary: should be ST & CN. Looks resolved now for Berserker, thanks.

**Comment by Sarsem** [comment]
Updated the Bard comment with new issues (all previously mentioned issues are resolved): would you prefer a new comment, or update for each class?

**Comment by gizmomathboy** [comment]
Just update as needed. I also don't need the pdf. 

Just give class, level, and what's wrong.

Thanks again for finding the probems

**Comment by Sarsem** [comment]
Cataphract: Skillful Defender doesn't appear accounted for, as Scale Armor + Shield is listed as AC 5, should be AC 3

**Comment by Sarsem** [comment]
Cleric: all issues resolved

**Comment by Sarsem** [comment]
Cryomancer: all issues resolved

**Comment by Sarsem** [comment]
Death Soldier: all issues resolved

**Comment by Sarsem** [comment]
Druid: all issues resolved

**Comment by Sarsem** [comment]
Fighter: all issues resolved

**Comment by Sarsem** [comment]
Fire Lord: all issues resolved

**Comment by Sarsem** [comment]
Fire Thief: all issues resolved

**Comment by gizmomathboy** [comment]
Wow, how did I miss the ROF changes for mastery at 7th+ level?

As for the !Law, !Good, should I just leave that blank? The intent was to show alignment restrictions if they existed. Should I just leave it blank?

I'll start work on the problems you've found.

**Comment by Sarsem** [comment]
I understand the alignment notation, not sure everyone will? Leaving it is fine, since the work is already done, and folks should know to replace it with an actual alignment.

**Comment by gizmomathboy** [comment]
for alignment I changed it to "Any" or the specific allowed alignments. 

It gets around my weird notation and such.

**Comment by Sarsem** [comment]
Sounds good

**Comment by gizmomathboy** [comment]
Cataphract should be ok.

I decided, per the comment about their starter pack, they have mastery with lance but won't have own. I add a lance to the starter pack for any cataphract greater than 1st level.

**Comment by gizmomathboy** [comment]
Spell casting stuff fixed.
Read scrolls fixed.

Now to look at ROF and the weapon mastery/weapon table stuff

**Comment by gizmomathboy** [comment]
ROF should now be fixed.

**Comment by gizmomathboy** [comment]
Ok, I think that death soldier, fire lord, and ice lord weapon mastery/weapon table stuff is fixed now.

**Comment by Sarsem** [comment]
Updated regression-tested classes: all appear good, will test the remaining over the next few days

**Comment by Sarsem** [comment]
Huntsman (though I don't think it's specific to this class): 9th-level had mastery in fighting net, hand axe and longbow, but short spear took up final Missile Weapon Skill row, and longbow wasn't shown. Probably want to list Mastered Weapons first, so they always appear?

**Comment by Sarsem** [comment]
Ice Thief: all issues resolved

**Comment by Sarsem** [comment]
Illusionist: all issues resolved

**Comment by gizmomathboy** [comment]
fixed the ice thief problem, but there still aren't enough slots depending on level.

illusionist fixed, but I think I uncovered a problem with dual school casters (bard/ranger/etc). I need to think on that one.

Same with the weapon mastery and weapon chart. Doable, I just need to think on it.

**Comment by gizmomathboy** [comment]
Ok, fixed the bonus spells for casters that use both schools of magic.

**Comment by gizmomathboy** [comment]
The weapon log stuff is gonna be tough. It's part a huge structural change I want to make. Not sure how long that will take to untangle

**Comment by Sarsem** [comment]
So for addressing running out of Class Abilities slots: why not replicate how the rulebook handles Weapon Mastery and Weapon Skill? I.e., list the ability once, and then in parentheses list the levels at which new mastery or skill is gained, since the player has to choose them anyway.

**Comment by gizmomathboy** [comment]
Good idea. The wording might not be the best but it works and saves some slots.

Now pick pockets shows up for level 12 ice thieves and such

**Comment by Sarsem** [comment]
Legerdemainist: all issues resolved

**Comment by gizmomathboy** [comment]
That was a weird one to leave out.

Fixed.

**Comment by Sarsem** [comment]
Monk: Defensive Ability should stack with DX bonus; Speak with Nature shows up twice

**Comment by Sarsem** [comment]
Spells: total number of spells known appears short of what is expected. E.g., a 9th level Priest should know 36 spells, but only 22 were listed; a 9th level Purloiner should know 18, only 7 were listed

**Comment by Sarsem** [comment]
Not class-specific, but noticed on Paladin: Large Shield should provide an overall +2 AC bonus, but was +1 vs Melee, +2 vs Missile

**Comment by gizmomathboy** [comment]
Got monk and large shields fixed.

I think I will need to pump up the default spells lists.

Also, I did a huge chunk of the back end stuff that I thought would be more difficult...but it wasn't quite that bad.

I don't think I've introduced errors...but this is code so it's inevitable.

**Comment by gizmomathboy** [comment]
To pad out default known spells I'm going to just randomly select them.

I can always revisit it later.

**Comment by Sarsem** [comment]
Random spells sounds good: as with any of the other fields they can always be edited.

I'll try and regression test some over the next few days: I did generate a 9th-level version of each remaining class which I have yet to assess.

**Comment by gizmomathboy** [comment]
I think the spell book stuff is working.

The only problem is priests. Not enough entries for their potential number of spells. Fills up at level 9.

I haven't looked at the 3e character sheet to see what it's like.

Also, the spells are listed alphabetically which follows how they are in the book.

**Comment by Sarsem** [comment]
Sounds good, I'll start generating 12th-level examples of each class (starting with Assassin again) over the next few days.
